ml/backend/ggml/ggml.go

```diff
@@ -1658,11 +1658,11 @@ func (t *Tensor) ScaledDotProductAttention(ctx ml.Context, key, value, mask, sin
 	if t.b.flashAttention {
 		value = value.Permute(ctx, 0, 2, 1, 3)
 
-		kqv := C.ggml_flash_attn_ext(ctx.(*Context).ctx, query.(*Tensor).t, key.(*Tensor).t, value.(*Tensor).t, kqMask, C.float(scale), 0, 0)
+		kqv := C.ggml_flash_attn_ext(ctx.(*Context).ctx, query.(*Tensor).t, key.(*Tensor).t, value.(*Tensor).t, kqMask, C.float(scale), 0, 0) // NOTE: kqv -> kqvC
```

Comment:
It looks like the flash attention path doesn't use vmla so presumably will be incorrect?

If there's a way to implement this using the existing attention mechanism and incorporate the extra parameter in the model definition, that would both fix the problem and keep the backend simpler.

If there's no way to do that and flash attention doesn't support this operation, then we would need to disable flash attention. However, as far as I know, the existing implementation in the llama engine doesn't have that restriction.